lib/dao.js

```diff
@@ -1661,7 +1665,8 @@ DataAccessObject._coerce = function(where) {
           operator = 'regexp';
         } else if (operator === 'regexp' && val instanceof RegExp) {
           // Do not coerce regex literals/objects
-        } else if (!((operator === 'like' || operator === 'nlike') && val instanceof RegExp)) {
+        } else if (!((operator === 'like' || operator === 'nlike' ||
+          operator === 'ilike' || operator === 'nilike') && val instanceof RegExp)) {
```

Comment:
The linter seems to fail to recognize indentation sometimes. See these lines:

>When not feasible, then indent the second and next lines by two levels.
>One level of indentation makes it difficult to tell the difference between the condition and the branch body.

Read the good/bad examples: https://github.com/strongloop/loopback-contributor-docs/blob/master/style-guide.md